engine/src/main/java/org/terasology/config/flexible/ui/FlexibleConfigTestScreen.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible.ui;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.config.flexible.FlexibleConfig;
+import org.terasology.config.flexible.FlexibleConfigManager;
+import org.terasology.config.flexible.Setting;
+import org.terasology.config.flexible.constraints.NumberRangeConstraint;
+import org.terasology.config.flexible.internal.FlexibleConfigManagerImpl;
+import org.terasology.engine.SimpleUri;
+import org.terasology.engine.module.ModuleManager;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.CoreScreenLayer;
+import org.terasology.rendering.nui.UILayout;
+import org.terasology.rendering.nui.widgets.UIButton;
+
+public class FlexibleConfigTestScreen extends CoreScreenLayer {
+    private static final SimpleUri CONFIG_ID = new SimpleUri("engine:TestConfig");
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlexibleConfigTestScreen.class);
+
+    private FlexibleConfigManager flexibleConfigManager = new FlexibleConfigManagerImpl();
+    private FlexibleConfig flexibleConfig;
+
+    private FlexibleConfigWidgetFactory configWidgetFactory;
+
+    private UILayout<?> mainContainer;
+
+    @In
+    ModuleManager moduleManager;
+
+    @In
+    AssetManager assetManager;
+
+    @Override
+    public void initialise() {
+        setupTestConfig();
+
+        configWidgetFactory = new FlexibleConfigWidgetFactory(moduleManager, assetManager);
+
+        mainContainer = find("mainContainer", UILayout.class);
+        assert mainContainer != null;
```

Comment:
Will this - and the other assert blocks - actually ever run? Assertions in Java are (perhaps counterintuitively) disabled by default, and I cannot find mentions of an `-enableassertions`/`-ea` switch in the codebase. Would consider replacing these with a simple exception check even if it's two lines longer.

Reply:
I don't believe assertions are enabled in Terasology. I've only used them here to satisfy IntelliJ, which otherwise complains about potential NPEs. I decided against using an actual null check, since this class is only ever meant to be used with the corresponding screen which will always have a "mainContainer" `UILayout`.